src/utils/commands.ts

```diff
@@ -0,0 +1,71 @@
+import { Message, Permissions } from 'discord.js';
+import Commando, { CommandoMessage, CommandoClient, CommandInfo, ArgumentCollectorResult } from 'discord.js-commando';
+import logger, { logError } from '../components/logger';
+
+export abstract class BaseCommand extends Commando.Command {
+  constructor(client: CommandoClient, info: CommandInfo) {
+    super(client, info);
+  }
+
+  run(
+    message: CommandoMessage,
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    args: object | string | string[],
+    fromPattern: boolean,
+    result?: ArgumentCollectorResult
+  ): Promise<Message | Message[] | null> | null {
+    // log each incoming command
+    onCommandRun(message);
+    return this.onRun(message, args, fromPattern, result);
+  }
+
+  onError(
+    err: Error,
+    message: CommandoMessage,
+    _args: unknown,
+    _fromPattern: boolean,
+    _result?: ArgumentCollectorResult | undefined
```

Comment:
ngl i dont get this file - too many _?:_:?_:?_:?_:?_:?_: lol

Reply:
the underscore just tells Typescript that we aren't using these args in the function, the question mark just means that this arg might not exist. I think we might be able to remove the `| undefined` here, let me check